### ISSUE TITLE
The Plan

### ISSUE THREAD

**Issue by DannyHatcher-Git** [blacksmithgu/datacore#1]
As someone that doesn't live on Github, seeing a plugin roadmap is really helpful.

The database folder plugin has a roadmap I often go to to check for future features, bugs, and plans.

Do you have any intention of including that here?
Just trying to get a sense of time here 😁

**Comment by blacksmithgu** [comment]
I'm still in "pre-production" where I'm scoping out the full extent of what will be supported, but I can set up a project kanban with the first public release in late January.

**Comment by DannyHatcher-Git** [comment]
Ok. I am excited to see what you come up with. 

It will be interesting as Obsidian has a metadata update planned, so there might be some overlap...

**Comment by Ravenclawer** [comment]
maybe this plugin should wait for the shipping of obsidian official metadata improvement feature
![image](https://user-images.githubusercontent.com/73892411/214472185-a85ffdd4-62bb-4211-95b1-e17c1cd2c450.png)

**Comment by Ktar5** [comment]
Is late january still slated to be the first public release? @blacksmithgu

**Comment by DannyHatcher-Git** [comment]
I was thinking maybe the team has asked if this could be the metadata improvements in core rather than a community plugin 🤷‍♂️

Update of the current search code block option...

**Comment by arminta7** [comment]
> I was thinking maybe the team has asked if this could be the metadata improvements in core rather than a community plugin 🤷‍♂️
> 
> 
> 
> Update of the current search code block option...

Interesting idea.

**Comment by imeed166** [comment]
> I was thinking maybe the team has asked if this could be the metadata improvements in core rather than a community plugin 🤷‍♂️
> 
> Update of the current search code block option...

That's why It's called data"core" as in "core plugin" xp

**Comment by geckom** [comment]
Could you please update the expected release date?

**Comment by blacksmithgu** [comment]
I've added a roadmap indicating the extent of the functionality I intend to support; I'm backing off a bit on the release date to "winter" which means sometime between now and mid-late-March.

**Comment by GamerGirlandCo** [comment]
is there anything i can do to help out? if so, what? i just don't want to accidentally open a PR for something you're already working on @blacksmithgu

**Comment by blacksmithgu** [comment]
> is there anything i can do to help out with developing this plugin? if so, what? i just don't want to accidentally open a PR for something you're already working on @blacksmithgu

@GamerGirlandCo There's a lot of ground to cover depending on your interest. I need a bit more time to put some critical pieces together, but I think coming up with + indexing canvas files would be really neat. I haven't thought through how the metadata for them should look like yet. Alternatively, adding some basic metadata extraction for PDF files, images, or videos.

**Comment by GamerGirlandCo** [comment]
> > is there anything i can do to help out with developing this plugin? if so, what? i just don't want to accidentally open a PR for something you're already working on @blacksmithgu
> 
> @GamerGirlandCo There's a lot of ground to cover depending on your interest. I need a bit more time to put some critical pieces together, but I think coming up with + indexing canvas files would be really neat. I haven't thought through how the metadata for them should look like yet. Alternatively, adding some basic metadata extraction for PDF files, images, or videos.

i've managed to implement a way to index canvas files with metadata (see [obsidian-dataview/#1803](https://github.com/blacksmithgu/obsidian-dataview/pull/1803)). let me know if there's anything you'd personally like to change, i'm open to improvements!

**Comment by daamiian** [comment]
I noticed the Obsidian team has moved "Metadata Improvements" back to _Short term_ status. 
Have you been in contact with the devs? I am waiting on both datacore and the core metadata improvements at this point.

**Comment by kngautham** [comment]
Are there any further updates on the release date? Thanks.

**Comment by GamerGirlandCo** [comment]
> Are there any further updates on the release date? Thanks.

soon™

**Comment by ondrobaco** [comment]
beautiful, can't wait for this plugin. Does querying at the the block/section level mean that a single file can be queried based on inline key/values? That is exactly what I'm looking for! When will this be out?

**Comment by xDovos** [comment]
hey,
i wanted to ask if there is an update on how long it might still take to release the plugin.
I'm asking because i'm wondering if it will release for obsidian october or not. 
the reason is that i'm currently creating a new example vault for dataview and related plugins but if it comes out this month then i might just wait for it and make it with datacore queries. (the general setup is already done, but it might need to change based on how the querying of lists and tasks changes in datacore).

also you might want to look at the "Modules" plugin. it solves the `require()` support for dataview and templater and is really powerful.

**Comment by GamerGirlandCo** [comment]
how would querying yaml work? i want to help implement this as it seems easy enough for me.

**Comment by blacksmithgu** [comment]
> hey, i wanted to ask if there is an update on how long it might still take to release the plugin. I'm asking because i'm wondering if it will release for obsidian october or not. the reason is that i'm currently creating a new example vault for dataview and related plugins but if it comes out this month then i might just wait for it and make it with datacore queries. (the general setup is already done, but it might need to change based on how the querying of lists and tasks changes in datacore).
> 
> also you might want to look at the "Modules" plugin. it solves the `require()` support for dataview and templater and is really powerful.

Sadly it didn't make it in time for Obsidian October :(

**Comment by blacksmithgu** [comment]
> how would querying yaml work? i want to help implement this as it seems easy enough for me.
> 
> is your intention to have YAML objects be "inline" fields (i.e., able to be placed inside lists and tasks)?

@GamerGirlandCo The first pass of YAML objects should be fairly straight forward- we just detect codeblocks that are annotated with `yaml` plus some datacore-specific flag (maybe `yaml,data` or something) and then add them as queryable objects underneath the section/page they are found in.

**Comment by GamerGirlandCo** [comment]
> > how would querying yaml work? i want to help implement this as it seems easy enough for me.
> > is your intention to have YAML objects be "inline" fields (i.e., able to be placed inside lists and tasks)?
> 
> @GamerGirlandCo The first pass of YAML objects should be fairly straight forward- we just detect codeblocks that are annotated with `yaml` plus some datacore-specific flag (maybe `yaml,data` or something) and then add them as queryable objects underneath the section/page they are found in.

Should YAML properties be accessible as inline fields?

**Comment by blacksmithgu** [comment]
I'm not decided, but I'm leaning towards no (or there should be a different annotation for that behavior). We already have properties and inline field syntax for that.

Maybe something like `yaml,inline` could mean that the block is parsed as inline fields (to allow for multi-line inline fields), but I'm not sure if the weirdness of the functionality is worth it.

**Comment by GamerGirlandCo** [comment]
![image](https://github.com/blacksmithgu/datacore/assets/37163052/3cb6eb90-022f-41da-a752-fe3998f41c05)
@blacksmithgu i made javascript codeblocks work :D

**Comment by Rhydderch** [comment]
Hi all,
Thank you for working on this project! I'm eagerly awaiting this to be implemented so I can now definitely convert notion people to obsidian haha. Just kidding, I'm excited about datacore because it will make obsidian even better (UI/UX wise). The latest properties update was already a big step in the right direction.

Related to @xDovos 's question: @blacksmithgu could you give us a very rough ETA of a release for datacore? And/Or would you consider updating the [roadmap](https://github.com/blacksmithgu/datacore/blob/master/ROADMAP.md) ?

(One of the reasons I am so interested in datacore is because I would like to create a template obsidian vault with github sync to enable research collaborations on theory building and literature reviews - datacore will definitely help with making summaries/overviews easier to manage!)

**Comment by blacksmithgu** [comment]
Yes - I think it's probably time to spin up a discord or other chat platform to make it easier for contributors to understand the State of The World (TM) and collaborate.

**Comment by error451** [comment]
does that mean the first phase where you worked on the core of the datacore is over and that the community will be involved from now on?

I'm looking forward to discord, this is for me the first pretty complex plugin that I want to "get into the code"
